common/scala/src/main/scala/whisk/core/database/MultipleReadersSingleWriterCache.scala

```diff
@@ -150,11 +153,13 @@ trait MultipleReadersSingleWriterCache[W, Winfo] {
      * This method posts a delete to the backing store, and either directly invalidates the cache entry
      * or informs any outstanding transaction that it must invalidate the cache on completion.
      */
-    protected def cacheInvalidate[R](key: Any, invalidator: => Future[R])(
+    protected def cacheInvalidate[R](key: CacheKey, invalidator: => Future[R], changeCacheCallback: CacheKey => Future[Unit] = CacheKey => Future.successful(Unit))(
```

Comment:
I thought about it, but I don't see a possibility to create the MessageProducer in the objects, that extend this trait, because of the missing executionContext.